File: python/metaparticle_pkg/docker_runner.py
import os
import subprocess

class DockerRunner:
    def run(self, img, name, options):
        # Launch docker container
        command = ['docker', 'run', '-d', '--name', name]
        if options.get('ports', None) is not None and len(options['ports']) > 0:
            command.append('-p')
            for i in options['ports']:
                command.append("{}:{}".format(i, i))
        command.append(img)
        subprocess.call(command)

    def logs(self, name):
        # Attach to logs
        # TODO: make this streaming...
        subprocess.call(['docker', 'logs', '-f', name])

    def cancel(self, name):
        subprocess.call(['docker', 'kill', name])
        subprocess.call(['docker', 'rm', name])

File: ruby/metaparticle/lib/metaparticle/docker_runner.rb
module Metaparticle
  class DockerRunner
    def initialize(config)
      @config = config
    end

    def run
      `docker run -it --name #{@config.name} #{port_string} #{@config.image}`
    end

    private
    def port_string
      @port_string = @config.ports.map {|port| "-p #{port}:#{port}" }.join(" ")
    end
  end
end
